src/test/scala/sigmastate/utxo/examples/DHTupleExampleSpecification.scala

```diff
@@ -0,0 +1,115 @@
+
+package sigmastate.utxo.examples
+
+import java.math.BigInteger
+
+import org.ergoplatform.ErgoBox.{R4, R5}
+import org.ergoplatform.{ErgoBox, ErgoLikeContext, ErgoLikeTransaction}
+import sigmastate.AvlTreeData
+import sigmastate.Values.GroupElementConstant
+import sigmastate.basics.DLogProtocol.ProveDlog
+import sigmastate.basics.{DiffieHellmanTupleProverInput, ProveDHTuple}
+import sigmastate.helpers.{ErgoLikeTestProvingInterpreter, SigmaTestingCommons}
+import sigmastate.interpreter.CryptoConstants
+import sigmastate.interpreter.Interpreter._
+import sigmastate.lang.Terms._
+import sigmastate.utxo.ErgoLikeTestInterpreter
+
+class DHTupleExampleSpecification extends SigmaTestingCommons {
+  private implicit lazy val IR = new TestingIRContext
+  /**
+    * let Alice's secret be x and Bob's be y
+    *
+    * Then
+    * (g, g^y, g^x, g^xy) forms a DH tuple for Alice and
+    * (g, g^x, g^y, g^xy) forms a DH tuple for Bob.
+    *
+    */
+  property("Evaluation - DH Example") {
+    import CryptoConstants.dlogGroup
+
+    val g = dlogGroup.generator
+
+    val alice = new ErgoLikeTestProvingInterpreter
+    val alicePubKey:ProveDlog = alice.dlogSecrets.head.publicImage
+
+    val x:BigInteger = alice.dlogSecrets.head.w // x is Alice's private key
+
+    val g_x = alicePubKey.h // g_x is Alice's public key (g_x = g^x)
+
+    val env = Map(
+      ScriptNameProp -> "env",
+      "g" -> g,
+      "g_x" -> g_x
+    )
+
+    val script = compileWithCosting(env,
+      """{
+        |  val g_y = OUTPUTS(0).R4[GroupElement].get
+        |  val g_xy = OUTPUTS(0).R5[GroupElement].get
+        |
+        |  proveDHTuple(g, g_x, g_y, g_xy) || // for bob
+        |  proveDHTuple(g, g_y, g_x, g_xy)    // for alice
+        |}""".stripMargin
+    ).asBoolValue
+
+    val inBox = ErgoBox(10, script, 50)
+
+    // a blockchain node verifying a block containing a spending transaction
+    val verifier = new ErgoLikeTestInterpreter
+
+    val bob = new ErgoLikeTestProvingInterpreter
+
+    val y:BigInteger = bob.dlogSecrets.head.w // y is Bob's private key
+
+    val g_y = GroupElementConstant(dlogGroup.exponentiate(g, y)) // g^y
```

Comment:
@scalahub Is there a reason to get Bob's public key this way? I find the following code easier to grasp and more in line with Alice's key's definitions:
```suggestion
   val bobPubKey: ProveDlog = bob.dlogSecrets.head.publicImage
   val g_y = GroupElementConstant(bobPubKey.h) // Bob's public key g^y
```
Good job explaining how DHT works, BTW!

Reply:
No particular reason. Just wanted to show how to generate a public key from a private value y. This version also fine.